web-extensions/chrome/engine/utils.js

```diff
@@ -0,0 +1,66 @@
+/**
+ *
+ * The FrameFinder class is an utility class which finds a frame from the
+ * web-extension based on the provided frame Definition.
+ * All Frame definitions are static members of the class, and this serves as a
+ * central point to reference all frames handled by the web extension.
+ *
+ * @class
+ */
+function FrameFinder() {
+}
+
+FrameFinder.definitions = {
+    SIDEPANEL: {
+        id: 'BmcSidePanel',
+        inspect: frame => !!(frame.showHideSidePanel),
+    },
+};
+
+/**
+ * This static method returns the first frame matching the provided frame
+ * Definition object (which should be one of the frameDefinitions owned by the
+ * class).
+ *
+ * @param {Object} frameDef - the frame Definition as definied by FrameFinder
+ *                            class
+ *
+ * @return {Frame} the DOM frame object matching the definition provided
+ */
+FrameFinder.findWindow = function(frameDef) {
+    if (!frameDef) {
+        alert('Cannot find an frame with unknown Definition');
+    }
+
+    let found = null;
+    if (window === window.top) {
+        // if we're in the top window, CORS will prevent introspecting the
+        // webext-owned frames, but we know the frame' ID
+
+        if (frameDef.id) {
+            found = document.getElementById(frameDef.id).contentWindow;
+            console.warn('FrameFinder.find: Fuond frame with id method');
+        }
+    } else {
+        // if we're not in the top window, then we're in a webext-owned frame,
+        // which should allow us to introspect our own frames, and find the
+        // sidepanel's
+        for (var i = 0; i < window.parent.length; ++i) {
+            let frame = window.parent.frames[i];
+            try {
+                if (frameDef.inspect && frameDef.inspect(frame)) {
+                    console.warn('FrameFinder.find: Fuond frame with inspect method');
+                    found = frame;
+                    break ;
+                }
+            } catch(e) {
+                continue ;
+            }
+        }
+    }
+    if (!found) {
```

Comment:
Remove this condition and just run the `alert`?

Reply:
indeed, with returns it can be done without extra conditionals.